File: moss-live-labs/typescript/advance_query.ts
import { MossClient } from '@moss-dev/moss'


import 'dotenv/config'
const client = new MossClient(process.env.MOSS_PROJECT_ID!, process.env.MOSS_PROJECT_KEY!)

const indexName = 'demo-customer_faqs'

async function main() {
  // ==========================load the index============================
  await client.loadIndex(indexName)
  console.log('Index loaded successfully.')

  // ==========================query the index============================
  const results = await client.query(indexName, 'How do I return a damaged product?', {
    topK: 3,
    alpha: 0.6,
    filter: { field: 'category', condition: { $eq: 'returns' } },
  })

  const doc = results.docs[0]
  console.log(`  ID: ${doc.id}`)
  console.log(`  Text: ${doc.text}`)
  console.log(`  Score: ${doc.score}`)
  console.log(`  Metadata: ${JSON.stringify(doc.metadata)}`)
  // =============================================================================
}

main()

File: moss-live-labs/python/advance_query.py
import os
import asyncio
from moss import MossClient, DocumentInfo, QueryOptions
from dotenv import load_dotenv
load_dotenv()




client = MossClient(os.getenv("MOSS_PROJECT_ID"), os.getenv("MOSS_PROJECT_KEY"))
index_name = "demo-customer_faqs"

async def main():
    #===========================load the index============================
    await client.load_index(index_name)
    print("Index loaded successfully.")

    #===========================query the index============================
    results = await client.query(
        index_name, 
        "How do I return a damaged product?", 
        QueryOptions(
            top_k=3, 
            alpha=0.6,
            filter={"field": "category", "condition": {"$eq": "returns"}}))
    
    print(f"  ID: {results.docs[0].id}")
    print(f"  Text: {results.docs[0].text}")
    print(f"  Score: {results.docs[0].score}")
    print(f"  Metadata: {results.docs[0].metadata}")
    #=============================================================================

asyncio.run(main())